csv_joiner.py

```diff
@@ -0,0 +1,147 @@
+import sys
+
+
+class InvalidModeError(Exception):
+    def __init__(self, modes):
+        self.expected = '/'.join(modes)
+        self.message = 'Invalid join mode'
+
+    def __str__(self):
+        return f'{self.message}. Expected: {self.expected}'
+
+
+class InvalidColumnError(Exception):
+    def __init__(self, col):
+        self.col = col
+        self.message = 'Column "{0}" does not appear in one/both files'
+
+    def __str__(self):
+        return self.message.format(self.col)
+
+
+class Parser:
+    @staticmethod
+    def parse_line(line: str) -> list:
+        line = line.rstrip()
+        start_i = 0
+        items = []
+        quotes = 0
+        for i, char in enumerate(line):
+            if char == ',' and not quotes:
```

Comment:
defining a small function naming the condition would be helpful for the reader, e.g
`if char == ',' and quotes_not_opened` or something even more descriptive